craft_platforms/snap/_build.py

```diff
@@ -0,0 +1,159 @@
+# This file is part of craft-platforms.
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Snapcraft-specific platforms information."""
+
+import itertools
+import re
+from collections.abc import Sequence
+
+from craft_platforms import _architectures, _buildinfo, _distro, _errors, _platforms
+
+CORE16_18_DEFAULT_ARCHITECTURES = (
+    _architectures.DebianArchitecture.AMD64,
+    _architectures.DebianArchitecture.ARM64,
+    _architectures.DebianArchitecture.ARMHF,
+    _architectures.DebianArchitecture.I386,
+    _architectures.DebianArchitecture.PPC64EL,
+    _architectures.DebianArchitecture.S390X,
+)
+
+CORE20_DEFAULT_ARCHITECTURES = (
+    _architectures.DebianArchitecture.AMD64,
+    _architectures.DebianArchitecture.ARM64,
+    _architectures.DebianArchitecture.ARMHF,
+    _architectures.DebianArchitecture.PPC64EL,
+    _architectures.DebianArchitecture.S390X,
+)
+
+DEFAULT_ARCHITECTURES_BY_BASE = {
+    "core": CORE16_18_DEFAULT_ARCHITECTURES,
+    "core16": CORE16_18_DEFAULT_ARCHITECTURES,
+    "core18": CORE16_18_DEFAULT_ARCHITECTURES,
+    "core20": CORE20_DEFAULT_ARCHITECTURES,
+}
+
+# The default architectures for the current and future releases.
+DEFAULT_ARCHITECTURES = (
+    _architectures.DebianArchitecture.AMD64,
+    _architectures.DebianArchitecture.ARM64,
+    _architectures.DebianArchitecture.ARMHF,
+    _architectures.DebianArchitecture.PPC64EL,
+    _architectures.DebianArchitecture.RISCV64,
+    _architectures.DebianArchitecture.S390X,
+)
+
+CORE_BASE_REGEX = re.compile("^core(16|18|[2-9][02468])?$")
+
+
+def get_default_architectures(base: str) -> Sequence[_architectures.DebianArchitecture]:
+    if base in DEFAULT_ARCHITECTURES_BY_BASE:
+        return DEFAULT_ARCHITECTURES_BY_BASE[base]
+    return DEFAULT_ARCHITECTURES
+
+
+def get_distro_base_from_core_base(
+    base: str,
```

Comment:
`base` is not always provided.  `base`, `kernel`, and `snapd` snaps are not allowed to define a base:
```yaml
name: core26
type: base
build-base: core24
```